framework/lib/components/combo-picker/combo-picker.tsx

```diff
@@ -30,15 +34,24 @@ export const ComboPicker = ({
   const buttonRef = useRef<any>()
   const selectRef = useRef<any>()
 
-  const getNewValue = (option: ComboPickerOption, input?: number): ComboPickerValue => ((input)
-    ? { input: Number(input), option }
-    : { option })
+  const getNewValue = (option: ComboPickerOption, input?: number): ComboPickerValue => (
+    (is(Number, input))
+      ? { input: Number(input), option }
+      : { option })
+
+  const getInputWithinRange = ((input: number) => {
+    if (max && input > max) return max
+    if (min && input < min) return min
+
+    return input
+  })
 
   const handleInputChange = (newInputvalue: NumberFormatValues) => {
     const newValue = getNewValue(selectOption, newInputvalue.floatValue)
 
     setInputValue(newValue.input)
-    onChange?.(newValue)
+    onChange?.({ ...newValue,
+      ...!is(Number, newValue.input) && defaultToZeroIfEmpty && { input: getInputWithinRange(0) } })
```

Comment:
Maybe put this in a separate variable? `const clampedDefaultValue = ...`

Reply:
Done!